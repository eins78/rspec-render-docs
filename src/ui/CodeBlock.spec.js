/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import CodeBlock from './CodeBlock'

const rubySrc = `
  def foo(bar)
    bar ? bar.to_foo : nil
  end
`

const jsonSrc = JSON.stringify({ foo: 'bar', baz: 1 }, 0, 2)

it('renders', () => {
  const component = renderer.create(
    <div>
      <CodeBlock l='ruby'>{rubySrc}</CodeBlock>
      <CodeBlock l='json'>{jsonSrc}</CodeBlock>
    </div>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
