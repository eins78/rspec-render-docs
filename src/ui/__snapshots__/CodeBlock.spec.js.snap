exports[`test renders 1`] = `
<div>
  <pre
    style={
      Object {
        "background": null,
        "color": "#333333",
        "display": "block",
        "overflowX": "auto",
        "padding": "0.5em",
      }
    }>
    <code>
      
        
      <span
        style={Object {}}>
        <span
          style={
            Object {
              "color": "#a71d5d",
            }
          }>
          def
        </span>
         
        <span
          style={
            Object {
              "color": "#795da3",
            }
          }>
          foo
        </span>
        <span
          style={Object {}}>
          (bar)
        </span>
      </span>
      
          bar ? bar.to_foo : 
      <span
        style={
          Object {
            "color": "#0086b3",
          }
        }>
        nil
      </span>
      
        
      <span
        style={
          Object {
            "color": "#a71d5d",
          }
        }>
        end
      </span>
      
      
    </code>
  </pre>
  <pre
    style={
      Object {
        "background": null,
        "color": "#333333",
        "display": "block",
        "overflowX": "auto",
        "padding": "0.5em",
      }
    }>
    <code>
      {
        
      <span
        style={
          Object {
            "color": "#795da3",
          }
        }>
        \"foo\"
      </span>
      : 
      <span
        style={
          Object {
            "color": "#df5000",
          }
        }>
        \"bar\"
      </span>
      ,
        
      <span
        style={
          Object {
            "color": "#795da3",
          }
        }>
        \"baz\"
      </span>
      : 
      <span
        style={Object {}}>
        1
      </span>
      
      }
    </code>
  </pre>
</div>
`;
